.github/workflows/linkcheck.yml

```diff
@@ -0,0 +1,104 @@
+name: Broken Links
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'source/**'
+      - 'poetry.lock'
+      - 'pyproject.toml'
+      - '.github/workflows/linkcheck.yml'
+  schedule:
+    # Nightly at 2 AM UTC
+    - cron: '0 2 * * *'
+  workflow_dispatch:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
It is good practice to Pin actions to a full-length commit SHA so that third-party actions cannot be compromised, I would recommend to do this on every new CI job from now on.